### ISSUE TITLE
jdk_nio_0_FAILED java/nio/file/attribute/BasicFileAttributeView/CreationTime.java#tmp UnsatisfiedLinkError: Native Library libCreationTimeHelper.so already loaded in another classloader

### ISSUE THREAD

**Issue by JasonFengJ9** [eclipse-openj9/openj9#19978]
Failure link
------------

From [an internal build](https://hyc-runtimes-jenkins.swg-devops.com/job/Test_openjdk21_j9_extended.openjdk_s390x_linux_testList_1/82/)(`sles15s390x-svl-rt4-1`):
```
java version "21.0.5-beta" 2024-10-15
IBM Semeru Runtime Certified Edition 21.0.5+1-202408031503 (build 21.0.5-beta+1-202408031503)
Eclipse OpenJ9 VM 21.0.5+1-202408031503 (build master-27dac39d2c, JRE 21 Linux s390x-64-Bit Compressed References 20240803_244 (JIT enabled, AOT enabled)
OpenJ9   - 27dac39d2c
OMR      - 9ccff3cf2
JCL      - fbac205a7 based on jdk-21.0.5+1)
```
[Rerun in Grinder](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/parambuild/?SDK_RESOURCE=customized&amp;TARGET=jdk_nio_0&amp;BASE_DOCKER_REGISTRY_CREDENTIAL_ID=&amp;TEST_FLAG=&amp;UPSTREAM_TEST_JOB_NAME=Test_openjdk21_j9_extended.openjdk_s390x_linux&amp;DOCKER_REQUIRED=false&amp;ACTIVE_NODE_TIMEOUT=5&amp;VENDOR_TEST_DIRS=&amp;EXTRA_DOCKER_ARGS=&amp;TKG_OWNER_BRANCH=adoptium%3Amaster&amp;OPENJ9_SYSTEMTEST_OWNER_BRANCH=eclipse%3Amaster&amp;PLATFORM=s390x_linux&amp;GENERATE_JOBS=false&amp;KEEP_REPORTDIR=true&amp;PERSONAL_BUILD=false&amp;DOCKER_REGISTRY_DIR=&amp;RERUN_ITERATIONS=0&amp;ADOPTOPENJDK_REPO=https%3A%2F%2Fgithub.com%2Fadoptium%2Faqa-tests.git&amp;SETUP_JCK_RUN=false&amp;DOCKER_REGISTRY_URL_CREDENTIAL_ID=&amp;LABEL=&amp;EXTRA_OPTIONS=&amp;CUSTOMIZED_SDK_URL=https%3A%2F%2Fna-public.artifactory.swg-devops.com%2Fartifactory%2Fsys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com%2Fbuild-scripts%2Fjobs%2Fjdk21u%2Fjdk21u-linux-s390x-openj9-IBM%2F244%2Fibm-semeru-certified-jdk_s390x_linux_JDK21U_2024-08-03-15-01.tar.gz+https%3A%2F%2Fna-public.artifactory.swg-devops.com%2Fartifactory%2Fsys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com%2Fbuild-scripts%2Fjobs%2Fjdk21u%2Fjdk21u-linux-s390x-openj9-IBM%2F244%2Fibm-semeru-certified-testimage_s390x_linux_JDK21U_2024-08-03-15-01.tar.gz&amp;BUILD_IDENTIFIER=&amp;JENKINS_KEY=Jenkins+4096+key&amp;ADOPTOPENJDK_BRANCH=master&amp;LIGHT_WEIGHT_CHECKOUT=false&amp;USE_JRE=false&amp;ARTIFACTORY_SERVER=na.artifactory.swg-devops&amp;KEEP_WORKSPACE=false&amp;USER_CREDENTIALS_ID=83181e25-eea4-4f55-8b3e-e79615733226&amp;JDK_VERSION=21&amp;DOCKER_REGISTRY_URL=&amp;ITERATIONS=1&amp;VENDOR_TEST_REPOS=&amp;JDK_REPO=git%40github.com%3Aibmruntimes%2Fopenj9-openjdk-jdk21&amp;JCK_GIT_BRANCH=main&amp;RELEASE_TAG=&amp;OPENJ9_BRANCH=master&amp;OPENJ9_SHA=&amp;JCK_GIT_REPO=&amp;VENDOR_TEST_BRANCHES=&amp;OPENJ9_REPO=https%3A%2F%2Fgithub.com%2Feclipse-openj9%2Fopenj9.git&amp;UPSTREAM_JOB_NAME=&amp;CLOUD_PROVIDER=&amp;CUSTOM_TARGET=&amp;VENDOR_TEST_SHAS=&amp;JDK_BRANCH=openj9&amp;LABEL_ADDITION=&amp;ARTIFACTORY_REPO=sys-rt-generic-local%2Chyc-runtimes-team-restricted-oracle-sdk-generic-local&amp;ARTIFACTORY_ROOT_DIR=&amp;UPSTREAM_TEST_JOB_NUMBER=119&amp;DOCKERIMAGE_TAG=&amp;TEST_TIME=&amp;JDK_IMPL=openj9&amp;SSH_AGENT_CREDENTIAL=83181e25-eea4-4f55-8b3e-e79615733226&amp;AUTO_DETECT=true&amp;SLACK_CHANNEL=%23rt-jenkins&amp;DYNAMIC_COMPILE=false&amp;RELATED_NODES=&amp;ADOPTOPENJDK_SYSTEMTEST_OWNER_BRANCH=adoptium%3Amaster&amp;APPLICATION_OPTIONS=&amp;CUSTOMIZED_SDK_URL_CREDENTIAL_ID=7c1c2c28-650f-49e0-afd1-ca6b60479546&amp;ARCHIVE_TEST_RESULTS=false&amp;NUM_MACHINES=&amp;OPENJDK_SHA=&amp;TRSS_URL=https%3A%2F%2Ftrssrtp1.fyre.ibm.com%2F&amp;USE_TESTENV_PROPERTIES=false&amp;BUILD_LIST=openjdk&amp;ADDITIONAL_ARTIFACTS_REQUIRED=&amp;UPSTREAM_JOB_NUMBER=&amp;STF_OWNER_BRANCH=adoptium%3Amaster&amp;TIME_LIMIT=25&amp;JVM_OPTIONS=&amp;PARALLEL=None) - Change TARGET to run only the failed test targets.

Optional info
-------------

Failure output (captured from console output)
---------------------------------------------
```
[2024-08-03T17:27:55.872Z] variation: Mode150
[2024-08-03T17:27:55.872Z] JVM_OPTIONS:  -XX:+UseCompressedOops -Xverbosegclog 

[2024-08-03T17:38:54.587Z] TEST: java/nio/file/attribute/BasicFileAttributeView/CreationTime.java#tmp

[2024-08-03T17:38:54.590Z] STDERR:
[2024-08-03T17:38:54.590Z] java.lang.UnsatisfiedLinkError: Native Library /home/jenkins/workspace/Test_openjdk21_j9_extended.openjdk_s390x_linux_testList_1/jdkbinary/openjdk-test-image/jdk/jtreg/native/libCreationTimeHelper.so already loaded in another classloader
[2024-08-03T17:38:54.590Z] 	at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:186)
[2024-08-03T17:38:54.590Z] 	at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:158)
[2024-08-03T17:38:54.590Z] 	at java.base/jdk.internal.loader.NativeLibraries.findFromPaths(NativeLibraries.java:278)
[2024-08-03T17:38:54.590Z] 	at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:270)
[2024-08-03T17:38:54.590Z] 	at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:1821)
[2024-08-03T17:38:54.590Z] 	at java.base/java.lang.System.loadLibrary(System.java:813)
[2024-08-03T17:38:54.590Z] 	at CreationTimeHelper.<clinit>(CreationTimeHelper.java:26)
[2024-08-03T17:38:54.590Z] 	at CreationTime.test(CreationTime.java:109)
[2024-08-03T17:38:54.590Z] 	at CreationTime.main(CreationTime.java:152)
[2024-08-03T17:38:54.590Z] 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
[2024-08-03T17:38:54.590Z] 	at java.base/java.lang.reflect.Method.invoke(Method.java:586)
[2024-08-03T17:38:54.590Z] 	at com.sun.javatest.regtest.agent.MainActionHelper$AgentVMRunnable.run(MainActionHelper.java:333)
[2024-08-03T17:38:54.590Z] 	at java.base/java.lang.Thread.run(Thread.java:1595)
[2024-08-03T17:38:54.590Z] 
[2024-08-03T17:38:54.590Z] JavaTest Message: Test threw exception: java.lang.UnsatisfiedLinkError
[2024-08-03T17:38:54.590Z] JavaTest Message: shutting down test
[2024-08-03T17:38:54.590Z] 
[2024-08-03T17:38:54.590Z] 
[2024-08-03T17:38:54.590Z] TEST RESULT: Failed. Execution failed: `main' threw exception: java.lang.UnsatisfiedLinkError: Native Library /home/jenkins/workspace/Test_openjdk21_j9_extended.openjdk_s390x_linux_testList_1/jdkbinary/openjdk-test-image/jdk/jtreg/native/libCreationTimeHelper.so already loaded in another classloader
[2024-08-03T17:38:54.590Z] --------------------------------------------------
[2024-08-03T17:48:19.943Z] Test results: passed: 408; failed: 1
[2024-08-03T17:48:57.295Z] Report written to /home/jenkins/workspace/Test_openjdk21_j9_extended.openjdk_s390x_linux_testList_1/aqa-tests/TKG/output_17227044527420/jdk_nio_0/report/html/report.html
[2024-08-03T17:48:57.295Z] Results written to /home/jenkins/workspace/Test_openjdk21_j9_extended.openjdk_s390x_linux_testList_1/aqa-tests/TKG/output_17227044527420/jdk_nio_0/work
[2024-08-03T17:48:57.295Z] Error: Some tests failed or other problems occurred.
[2024-08-03T17:48:57.295Z] -----------------------------------
[2024-08-03T17:48:57.295Z] jdk_nio_0_FAILED
```

[50x internal grinder](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/42635/) - 25/50 failed

Also seen at [JDK21 x86-64_linux](https://hyc-runtimes-jenkins.swg-devops.com/job/Test_openjdk21_j9_extended.openjdk_x86-64_linux_testList_1/96/)
[JDK17 ppc64le_linux](https://hyc-runtimes-jenkins.swg-devops.com/job/Test_openjdk17_j9_extended.openjdk_ppc64le_linux_testList_0/111/)

**Comment by gacholio** [comment]
This kind of failure usually means that the test expected the class loader to be unloaded so that the subsequent loader could load the library. Class loader unloading is non-deterministic, so if this is the case, the test is simply invalid.

**Comment by JasonFengJ9** [comment]
This seems a new test behaviour introduced via https://github.com/ibmruntimes/openj9-openjdk-jdk21/commit/2d5b7d581f77d82fa8c2b0be8524445bc74720e5
Tried an earlier openj9 build which [failed](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/42653/consoleFull) as well.
```
18:54:04  java version "21.0.4-beta" 2024-07-16
18:54:04  IBM Semeru Runtime Certified Edition 21.0.4+7-202407240348 (build 21.0.4-beta+7-202407240348)
18:54:04  Eclipse OpenJ9 VM 21.0.4+7-202407240348 (build master-9d60105111, JRE 21 Linux s390x-64-Bit Compressed References 20240724_236 (JIT enabled, AOT enabled)
18:54:04  OpenJ9   - 9d60105111
18:54:04  OMR      - 56cb81f5e
18:54:04  JCL      - b473f9fdf based on jdk-21.0.4+7)
```
RI [passed](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/42652/consoleFull) however RI might have different class loader unloading implementations.

Removing this from 0.48 milestone plan.

**Comment by gacholio** [comment]
I thought maybe we had implemented a workaround for this (GC if we fail to load a library for this reason), but maybe not. It's certainly something we could consider.

**Comment by pshipton** [comment]
@tajila fyi a deficiency exposed by a new test.

**Comment by JasonFengJ9** [comment]
> This kind of failure usually means that the test expected the class loader to be unloaded so that the subsequent loader could load the library. Class loader unloading is non-deterministic, so if this is the case, the test is simply invalid.

This test doesn't intend to test `System.loadLibrary()` behaviour for the same native library. The `UnsatisfiedLinkError` was because the classloader wasn't unloaded in time before the second `System.loadLibrary()`.

It is not a VM issue.
Do we consider this a GC issue, or a test not applying to OpenJ9?

**Comment by pshipton** [comment]
If it can be fixed as suggested in https://github.com/eclipse-openj9/openj9/issues/19978#issuecomment-2278666732 then it's VM.

**Comment by JasonFengJ9** [comment]
Okay, I see what GAC meant.

**Comment by babsingh** [comment]
@JasonFengJ9 For 0.48, this issue will need to be resolved by the end of this week. What's the current state of this issue? Based on this issue's impact, do we need it to be fixed in 0.48 or can it be pushed to 0.49?

**Comment by JasonFengJ9** [comment]
This is a long-existing issue, it isn't required for 0.48.

Moving to 0.49.
